lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -933,7 +933,7 @@ function dosomething_campaign_get_goals_times($node) {
   $timing = [
     "today" => new DateTime(),
     "start_date" => new DateTime($high_season[0]['value']),
-    "end_date" => new DateTime($high_season[0]['value2']),
+    "end_date" => date_modify(new DateTime($high_season[0]['value2']), '+23 hours +59 minutes +59 seconds'),
```

Comment:
Just seemed cleaner and more reliable. Also looking at it should hopefully make more sense "we are adding 23 hours 59 minutes and seconds to this date object".